lib/libutee/tee_api_objects.c

```diff
@@ -109,15 +104,10 @@ TEE_Result TEE_RestrictObjectUsage1(TEE_ObjectHandle object, uint32_t objectUsag
 
 	res = utee_cryp_obj_restrict_usage((uint32_t)object, objectUsage);
 
-	if (res == TEE_ERROR_CORRUPT_OBJECT) {
-		res = utee_storage_obj_del(object);
-		if (res != TEE_SUCCESS)
-			TEE_Panic(0);
-		return TEE_ERROR_CORRUPT_OBJECT;
-	}
-
-	if (res != TEE_SUCCESS && res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
-		TEE_Panic(0);
+	if (res != TEE_SUCCESS &&
+	    res != TEE_ERROR_CORRUPT_OBJECT &&
```

Comment:
Other `utee_storage_obj_del()` should be removed, such as the one in `TEE_GetObjectInfo1()`. According to me, `utee_storage_obj_del()` should be only called from `TEE_CloseAndDeletePersistentObject1()`, all the other "remove" being done automatically in the core.